Source/Core/VideoCommon/VertexShaderGen.cpp

```diff
@@ -282,20 +340,16 @@ static T GenerateVertexShader(API_TYPE api_type)
 				break;
 		}
 
-		uid_data->dualTexTrans_enabled = xfmem.dualTexTrans.enabled;
 		// CHECKME: does this only work for regular tex gen types?
-		if (xfmem.dualTexTrans.enabled && texinfo.texgentype == XF_TEXGEN_REGULAR)
+		if (uid_data->dualTexTrans_enabled && texinfo.texgentype == XF_TEXGEN_REGULAR)
 		{
-			const PostMtxInfo& postInfo = xfmem.postMtxInfo[i];
+			auto& postInfo = uid_data->postMtxInfo[i];
 
-			uid_data->postMtxInfo[i].index = xfmem.postMtxInfo[i].index;
-			int postidx = postInfo.index;
 			out.Write("float4 P0 = " I_POSTTRANSFORMMATRICES"[%d];\n"
 				"float4 P1 = " I_POSTTRANSFORMMATRICES"[%d];\n"
 				"float4 P2 = " I_POSTTRANSFORMMATRICES"[%d];\n",
-				postidx & 0x3f, (postidx + 1) & 0x3f, (postidx + 2) & 0x3f);
+					  postInfo.index & 0x3f, (postInfo.index + 1) & 0x3f, (postInfo.index + 2) & 0x3f);
```

Comment:
Weird mixture of tabs and spaces here.
Doesn't line up with the previous lines that relate to it.